sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -73,17 +73,7 @@ class JDBCOptions(
   // ------------------------------------------------------------
   // Optional parameters
   // ------------------------------------------------------------
-  val driverClass = {
-    val userSpecifiedDriverClass = parameters.get(JDBC_DRIVER_CLASS)
-    userSpecifiedDriverClass.foreach(DriverRegistry.register)
-
-    // Performing this part of the logic on the driver guards against the corner-case where the
-    // driver returned for a URL is different on the driver and executors due to classpath
-    // differences.
-    userSpecifiedDriverClass.getOrElse {
-      DriverManager.getDriver(url).getClass.getCanonicalName
-    }
-  }
+  val driverClass = parameters.get(JDBC_DRIVER_CLASS)
```

Comment:
```Scala
spark.read.format("jdbc").option("url", "jdbc:sqlite:").option("dbtable", "x").load
```
In the above code, it is not passing `JDBC_DRIVER_CLASS`. Thus, this does not help. 

Actually, if @darabos passes `JDBC_DRIVER_CLASS` as an JDBC option, the problem is solved.  Thus, I suspect it is caused by the classloader.